Hybrid oil heat pumps in ETM

KoenvanB opened the pull request:
This PR adds oil-based hybrid heat pumps to the front end. It includes the following additions:

1. Oil HHP slider for households and for buildings;
2. Threshold COP slider for oil HHP

The PR also includes the following changes to existing material:

- Updates of the `Demand response - behavior of hybrid heat pumps` slide text to refer to general HHPs (instead of only gas-based ones)
- Update of threshold COP slider texts for space heating & for hot water with additional explanation;
- Hides the `hybrid_heat_pump` table for electricity & gas shares (see [corresponding new issue](https://github.com/quintel/etmodel/issues/4237))
- label update for HHP gas & electricity consumer prices (closes [#4219](https://github.com/quintel/etmodel/issues/4219))

mabijkerk commented on config/locales/interface/input_elements/en_demand_buildings.yml:
```diff
@@ -473,3 +473,37 @@ en:
+      title: Hybrid air heat pump (oil)
+      short_description:
+      description: |
+        This heater is a combination of an electric heat pump that draws its heat 
```
I feel like the description for the HHP (or for all HHPs) is very long. I would prefer to keep the slider description short and to the point with relevant references to charts/other parts of the model, whereas the documentation may contain a more elaborate explanation of HHPs.

mabijkerk commented on config/locales/interface/input_elements/en_demand_buildings.yml:
```diff
@@ -473,3 +473,37 @@ en:
+        from the outside air and an efficient conventional oil-fired boiler. <br/><br/>
+        This works as follows: <br/> 
+        The oil-fired part takes over all of the heating of the building on cold 
+        winter days, when heat demand peaks. Because of this clever trick, the heat 
+        pump has less impact on the electricity grid than an all-electric heat pump does. 
+        Smart use of hybrid crude oil heat pumps may therefore avoid the need to reinforce 
+        local power grids. <br/><br/>
```
Referring to the behaviour of the HHP as a "clever trick" does not seem very professional.

mabijkerk commented:
For e.g., Germany or Northern Ireland, setting space heating 100% to hybrid (oil) heat pumps does not actually turn on the 'hybrid' part. By default, no ambient heat is consumed. The reason for this is that the threshold COP is rather high and decreasing it does lead to ambient heat consumption.

The functionaliteit is therefore verified, but in terms of validation I do wonder whether this is expected behaviour @KoenvanB? This effectively means that under current (or 2019) price conditions and temperatures, the hybrid heat pump will functions as an oil heater. Have you checked this with the client?

mabijkerk commented:
> Hides the hybrid_heat_pump table for electricity & gas shares (see https://github.com/quintel/etmodel/issues/4237)

This table is still available through the slide text at Demand response - behavior of hybrid heat pumps":
<img width="465" alt="Screenshot 2024-04-09 at 12 24 49" src="https://github.com/quintel/etmodel/assets/67338510/9b90dc32-01d5-48a1-be79-ddbf57ca50f4">

KoenvanB commented:
I have 
- shortened the slide & slider texts by including part of it in the documentation (and inserted a reference to the documentation);
- removed the link to the hidden table;
- moved the oil HHP slider position to sit below the hydrogen HHP slider

KoenvanB commented:
With the alterations mentioned adjustments this branch is ready to merge!
